package.json

```diff
@@ -11,53 +11,63 @@
     "fetch"
   ],
   "packageManager": "pnpm@7.1.0",
-  "main": "./core/dist/index.js",
-  "module": "./core/dist/index.esm.js",
-  "types": "./core/dist/index.d.ts",
+  "main": "./dist/index.js",
+  "module": "./dist/index.mjs",
+  "types": "./dist/index.d.ts",
+  "typesVersions": {
+    "*": {
+      ".": [
+        "./dist/index.d.ts"
+      ],
+      "./infinite": [
+        "./dist/infinite.d.ts"
+      ],
+      "./immutable": [
+        "./dist/immutable.d.ts"
+      ],
+      "./mutation": [
+        "./dist/mutation.d.ts"
+      ],
+      "./_internal": [
+        "./dist/_internal.d.ts"
+      ]
+    }
+  },
   "exports": {
     "./package.json": "./package.json",
     ".": {
-      "types": "./core/dist/index.d.ts",
-      "import": "./core/dist/index.mjs",
-      "module": "./core/dist/index.esm.js",
-      "require": "./core/dist/index.js"
+      "types": "./dist/index.d.ts",
+      "import": "./dist/index.mjs",
+      "module": "./dist/index.mjs",
+      "require": "./dist/index.js"
     },
     "./infinite": {
-      "types": "./infinite/dist/index.d.ts",
-      "import": "./infinite/dist/index.mjs",
-      "module": "./infinite/dist/index.esm.js",
-      "require": "./infinite/dist/index.js"
+      "types": "./dist/infinite.d.ts",
+      "import": "./dist/infinite.mjs",
+      "module": "./dist/infinite.mjs",
```

Comment:
I'm not sure why [bunchee](https://github.com/huozhi/bunchee) currently can't produce output with `.esm.js`  extensions.  But we could fix this later

Reply:
"module" field with ".mjs" extension breaks react native bundler metro before, so we keep a different extension here.